### PR TITLE
Add ACME docs and link to it from README and INSTALL

### PR DESCRIPTION
**Pull request by anoadragon453**
Make docs clearer about what's necessary to upgrade to v0.99.0 depending on your cert situation.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -84,6 +84,13 @@ Synapse Installation
 For details on how to install synapse, see `<INSTALL.md>`_.
 
 
+To actually run your new homeserver, pick a working directory for Synapse to
```
Comment (richvdh) [inline README.rst]: this is only correct if you installed from source; I moved it out deliberately

File: README.rst [hunk 2]
```diff
@@ -127,6 +134,12 @@ create the account. Your name will take the form of::
 As when logging in, you will need to specify a "Custom server".  Specify your
 desired ``localpart`` in the 'User name' box.
 
+ACME setup
```
Comment (richvdh) [inline README.rst]: doesn't really belong under "Connecting to Synapse from a client" imho

File: INSTALL.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ pip install matrix-synapse[all]
 
 This will download Synapse from [PyPI](https://pypi.org/project/matrix-synapse)
 and install it, along with the python libraries it uses, into a virtual environment
-under ``~/synapse/env``.  Feel free to pick a different directory if you
```
Comment (richvdh) [inline INSTALL.md]: ftr double-backticks seem to mostly work in md; cleaning them up is good though

File: UPGRADE.rst [hunk 4]
```diff
@@ -51,34 +51,10 @@ returned by the Client-Server API:
...
-
-You will need to back up or delete your self signed TLS certificate
-(``example.com.tls.crt`` and ``example.com.tls.key``), Synapse's ACME
-implementation will not overwrite them.
-
-You may wish to use alternate methods such as Certbot to obtain a certificate
-from Let's Encrypt, depending on your server configuration. Of course, if you
-already have a valid certificate for your homeserver's domain, that can be
-placed in Synapse's config directory without the need for ACME.
+No special steps are required, but please be aware that you will need to
```
Comment (richvdh) [inline UPGRADE.rst]: suggest "please be aware that, before Synapse v1.0 is released around in March 2019, you will need to ... "

File: docs/ACME.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
+# ACME
+
+Synapse v1.0 requires that federation TLS certificates are verifiable by a
```
Comment (richvdh) [inline docs/ACME.md]:
```suggestion
Synapse v1.0 will requirs that federation TLS certificates are verifiable by a
```

File: docs/ACME.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+trusted root CA. If you do not already have a valid certificate for your domain, the easiest
+way to get one is with Synapse's new ACME support, which will use the ACME
```
Comment (richvdh) [inline docs/ACME.md]:
```suggestion
way to get one is with Synapse's ACME support (new as of Synapse 0.99), which will use the ACME
```

File: docs/ACME.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+them.
+
+You may wish to use alternate methods such as Certbot to obtain a certificate
+from Let's Encrypt, depending on your server configuration. Of course, if you
+already have a valid certificate for your homeserver's domain, that can be
+placed in Synapse's config directory without the need for any ACME setup.
+
+## ACME setup
+
+Synapse v1.0 will require valid TLS certificates for communication between servers
```
Comment (richvdh) [inline docs/ACME.md]: we need to combine this with the introduction - it basically says the same thing.

File: docs/ACME.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+sudo chmod 777 /etc/authbind/byport/80
+```
+
+When Synapse is started, use the following syntax::
+
+```
+authbind --deep <synapse start command>
+```
+
+Finally, once Synapse is able to listen on port 80 for ACME challenge
```
Comment (richvdh) [inline docs/ACME.md]: need to do this for reverse-proxy mode too

File: README.rst (hunk 1 above)
Comment (anoadragon453) [reply README.rst]: kk. It's also in INSTALL.md, should I leave it there?

File: docs/ACME.md (hunk 8 above)
Comment (anoadragon453) [reply docs/ACME.md]: Yeah, it was meant as an ending to both. I'll add another heading.

**Pull request merged**